tests_e2e.py

```diff
@@ -0,0 +1,780 @@
+        self.post_data('programs/share', {'id': program_id, 'public': False})
+
+        saved_programs = self.get_data('programs_list')['programs']
+        for program in saved_programs:
+            if program['id'] != program_id:
+                continue
+            # THEN the program must have no `public` field
+            self.assertNotIn('public', program)
+
+        # GIVEN another user
+        self.given_fresh_user_is_logged_in()
+        # WHEN requesting a public program
+        # THEN receive a not found response code from the server
+        self.get_data('hedy/1/' + program_id, expect_http_code=404)
+
+    def test_invalid_delete_program(self):
+        # GIVEN a logged in user with at least one program
+        self.given_user_is_logged_in()
+        program = {'code': 'hello world', 'name': 'program 1', 'level': 1}
+        self.post_data('programs', program)['id']
+        program_id = '123456'
+
+        # WHEN deleting a program that does not exist
+        # THEN receive a not ound response code from the server
+        self.get_data('programs/delete/' + program_id, expect_http_code=404)
+
+    def test_valid_delete_program(self):
+        # GIVEN a logged in user with at least one program
+        self.given_user_is_logged_in()
+        program = {'code': 'hello world', 'name': 'program 1', 'level': 1}
+        program_id = self.post_data('programs', program)['id']
+
+        # WHEN deleting a program
+        # THEN receive an OK response code from the server
+        headers = self.get_data('programs/delete/' + program_id, expect_http_code=302, return_headers=True)
+        # THEN verify that the header has a `location` header pointing to `/programs`
+        self.assertEqual(headers['location'], HOST + 'programs')
+
+        saved_programs = self.get_data('programs_list')['programs']
+        for program in saved_programs:
+            # THEN the program should not be any longer in the list of programs
+            self.assertNotEqual(program['id'], program_id)
+
+    def test_destroy_account_with_programs(self):
+        # GIVEN a logged in user with at least one program
+        self.given_user_is_logged_in()
+        program = {'code': 'hello world', 'name': 'program 1', 'level': 1}
+        program_id = self.post_data('programs', program)['id']
+
+        # WHEN deleting the user account
+        # THEN receive an OK response code from the server
+        self.post_data('auth/destroy', '')
+
+        # FINALLY remove user since it has already been deleted in the server
+        USERS.pop(self.username)
+
+# *** CLEANUP ***
+
+# We delete all the test users we created during the tests.
+# For this purpose, we use a pytest fixture. This requires us to use the `request` variable, without any possible renaming.
+# For this reason, we must rename our `request` function to `Request` so it will be referenceable from within the fixture.
+Request = request
+@pytest.fixture(scope='session', autouse=True)
```

Comment:
This is pytest-specific. If that is necessary, then I guess it will just have to do but for flexibility I was hoping to keep the code restricted to `unittest` if possible.

Does [tearDownModule](https://docs.python.org/2/library/unittest.html#setupmodule-and-teardownmodule) not work?